src/hera/workflow.py

```diff
@@ -320,3 +327,40 @@ def get_parameter(self, name: str) -> Parameter:
         if self.parameters is None or next((p for p in self.parameters if p.name == name), None) is None:
             raise KeyError(f"`{name}` is not a valid workflow parameter")
         return Parameter(name, value=f"{{{{workflow.parameters.{name}}}}}")
+
+    def to_dict(self) -> dict:
+        """Returns the JSON/dictionary representation of the workflow"""
+        return self.build().to_dict()
+
+    def to_yaml(self) -> str:
+        """Returns a YAML representation of the workflow"""
+        dict_repr = self.to_dict()
+        new_dict_repr: dict = {}
+        _traverse(dict_repr, new_dict_repr, _camel_case)
```

Comment:
This will unfortunately also camel-case keys that should be preserved. For example keys in `metadata.labels`, `metadata.annotations`, `spec.nodeSelector` and many more. In general terms, I think we need to preserve keys on generic dictionaries and only camel-case keys that are actual object fields. I'm not much of a Python developer, so not sure about the implementation.

Reply:
That is a very good point, thank you for bringing it up. I will adjust the implementation!

Reply:
Ended up going with @samj1912 suggestion and now fields like labels, annotations, and selectors are correctly ignored while everything else is parsed